utils/s2n_blob.c

```diff
@@ -28,7 +28,10 @@ S2N_RESULT s2n_blob_validate(const struct s2n_blob *b)
     RESULT_ENSURE_REF(b);
     RESULT_DEBUG_ENSURE(S2N_IMPLIES(b->data == NULL, b->size == 0), S2N_ERR_SAFETY);
     RESULT_DEBUG_ENSURE(S2N_IMPLIES(b->data == NULL, b->allocated == 0), S2N_ERR_SAFETY);
-    RESULT_DEBUG_ENSURE(S2N_IMPLIES(b->growable == 0, b->allocated == 0), S2N_ERR_SAFETY);
+    /* a blob is not growable <-> a blob is not allocated */
+    RESULT_DEBUG_ENSURE(S2N_IFF(b->growable == 0, b->allocated == 0), S2N_ERR_SAFETY);
+    /* a blob is allocated <-> a blob is growable */
+    RESULT_DEBUG_ENSURE(S2N_IFF(b->allocated > 0, b->growable == 1), S2N_ERR_SAFETY);
```

Comment:
Isn't the point of "IFF" that you only have to write it once? Unless I'm missing something, you're saying "growable == allocated && allocated == growable". Like, what situation does line 32 not catch?

Reply:
🤦 

It's been too long since my logic course.